include/gzip/config.hpp

```diff
@@ -0,0 +1,5 @@
+#pragma once
+
```

Comment:
The idea here is to define the `ZLIB_CONST` in one place. Hence this new header.

Reply:
Is the idea to use this for other things as well? Like WERROR, for example?